src/main/java/picocli/CommandLine.java

```diff
@@ -18972,11 +18972,25 @@ private static Properties loadProperties(CommandSpec commandSpec) {
             if (commandSpec == null) { return null; }
             Properties p = System.getProperties();
             for (String name : commandSpec.names()) {
+                String propertiesFileName = "." + name + ".properties";
                 String path = p.getProperty("picocli.defaults." + name + ".path");
-                File defaultPath = new File(p.getProperty("user.home"), "." + name + ".properties");
+                File defaultPath = new File(p.getProperty("user.home"), propertiesFileName);
                 File file = path == null ? defaultPath : new File(path);
                 if (file.canRead()) {
                     return createProperties(file, commandSpec);
+                } else {
+                    URL resource = commandSpec.userObject().getClass().getClassLoader().getResource(propertiesFileName);
```

Comment:
Minor point here is that the commandSpec's `userObject` may be `null` in some circumstances.
In that case we have little choice but to fall back to the class loader of the `picocli.CommandLine` class.
Can I ask you to make that change?